gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/ExecutionTreeBuilder.java

```diff
@@ -357,4 +356,54 @@ private QueryNode createQsDataFetcherNodeWithLimitAndOffset(EntitiesRequest enti
   private QueryNode createPaginateOnlyNode(QueryNode queryNode, EntitiesRequest entitiesRequest) {
     return new PaginateOnlyNode(queryNode, entitiesRequest.getLimit(), entitiesRequest.getOffset());
   }
+
+  private Optional<String> getSingleSourceForAllAttributes() {
+    // All expressions' attributes from the same source. Will only need one downstream query.
+    Optional<String> singleSourceForAllAttributes =
+        ExecutionTreeUtils.getSingleSourceForAllAttributes(executionContext);
```

Comment:
We have two methods with the same name, but different behavior. Is the other one called in other places? Does this one belong in the utils file? If I were new to this code (which I am!) I wouldn't know which to use when.

Reply:
Moved into a single method inside the utils